mysql-test/lib/mtr_cases.pm

```diff
@@ -639,7 +639,7 @@ sub optimize_cases {
     # Check that engine selected by
     # --default-storage-engine=<engine> is supported
     # =======================================================
-    my %builtin_engines = ('myisam' => 1, 'memory' => 1, 'csv' => 1);
+    my %builtin_engines = ('myisam' => 1, 'memory' => 1, 'csv' => 1, 'rocksdb' => 1);
```

Comment:
I think this is indeed the collect_mysqld_features issue, not necessarily the test issue itself. MTR runs collect_mysqld_features, does not see MyRocks there, does not set rocksdb as available, and as a result `--mysqld=--default-storage-engine=rocksdb` considers rocksdb as unavailable. With the fixes above `--mysqld=--default-storage-engine=tokudb --mysqld=--plugin-load=ha_tokudb.so=blahblahblah` works some of the time for some configurations, but it's not good enough.